### PR TITLE
The Magical Realms of Tír na nÓg: Escape from Necron 7 – Revenge of Cuchulainn: The Official Game of the Movie – Chapter 2 of the Hoopz Barkley Saga

### PR DESCRIPTION
**Pull request by McDonald072**
Read the convo to get an idea what this is I'm too lazy to explain

:cl:
rscadd: Strange reports of stations being torn asunder with basketballs have begun appearing in your sector. Watch out for personnel attempting to access the vault while holding a b-ball.
/:cl:

### CODE REVIEW

**Comment by ShadowDeath6** [comment]
so when is yog offically becoming the hub for spacejam spinoff game lore?

**Comment by Slinky5890** [comment]
You need to make a way for them to not go hide in lockers or something. I can also just imagine someone walling an area off and it taking 60 seconds for someone to get tools/RCD and come back. Maybe increase it to 5 minutes or something and have it so they can't stay still. Dunno how possible that last bit is

**Comment by ShadowDeath6** [comment]
I'll probably make it so syndie toolboxes hold the special screwdrivers so traitors with glorious death can get it because face it some ~shithead~ admin is going to ~bitch~ pm you about murderbone when you memeblast everyone instead of greentexting

**Comment by 1fbff5f83b23d39d38b1dfcb4cac8d9b** [comment]
It's supposed to be SaGa shadow 

I'm really torn on this. I want it to be possible only under rare circumstances, but glorious death people would really benefit from it, aaaaahhhh... 

Thanks slinky, those are really important things to consider

**Comment by Time-Green** [comment]
Also take in mind it's not fun to start a chaos dunk and then get disarmed by 20 assistants

**Comment by 1fbff5f83b23d39d38b1dfcb4cac8d9b** [comment]
Git gud and dribble away, this is what the challenge is about

**Comment by Time-Green** [comment]
Just dont make it 5 minutes

**Comment by McDonald072** [comment]
Okay so the plan is that a spirit hoop will appear somewhere on the station and you'll have to make your way over to it without getting disarmed by an assisstant

**Comment by ShadowDeath6** [comment]
>needing a hoop

:

**Comment by ghost** [comment]
Doing this will still get you banned, since you can never get this and die a glorious death at the same time.

The sole exception would probably be wizards, or if a hijack traitor can somehow rig up an autocloner away from the blast to come back after the station evaporates and he dies.

Obviously, I hate the fact that this will get you banned, and am updating toolboxes accordingly.

**Comment by McDonald072** [comment]
finally adam is making a change everyone can agree with

**Comment by fonedsky** [comment]
If you don't merge this and Adam's PR I swear to god I will find every single code maintainer.

File: code/modules/holodeck/items.dm [hunk 1]
```diff
@@ -94,6 +94,17 @@
...
+
+/obj/item/toy/beach_ball/holoball/chaos/attack(mob/living/M, mob/user)
+	if(!jamming)
+		return
+	if(istype(M))
+		M.Weaken(2)
+		user.do_attack_animation(M)
+		playsound(get_turf(user), 'sound/effects/hit_kick.ogg', 100, 0)
+		user.visible_message("<span class='danger'>[user] slams [M] onto the ground!</span>")
+		user.changeNext_move(CLICK_CD_MELEE) //Who the fuck mixed camelcase with snakecase?
```
Comment (Cruix) [inline code/modules/holodeck/items.dm:120]: Chamelcase requires lowercase letters

Comment (McDonald072) [reply code/modules/holodeck/items.dm:120]: changeNext seems pretty camelCase to me

Comment (Cruix) [reply code/modules/holodeck/items.dm:120]: I thought you were talking about the define

**Comment by McDonald072** [comment]
Note: If an admin prevents a chaos dunk, then revives the person, they will become unstunnable, but they still can take damage. ????

File: code/modules/holodeck/items.dm [hunk 2]
```diff
@@ -111,10 +128,40 @@
...
+		barkleys_shorts.flags = NODROP
+		H.equip_to_appropriate_slot(barkleys_shorts)
+		H << "<span class='danger'>You call forth the spirit of Barkley and begin channeling his power...</span>"
+		H << "<span class='danger'>Show off your dribbling skills and evade your opponents!<span>"
+		while(!slam_ready) //I feel like this is a really bad way to code this
+			if(H.get_active_hand() != src && H.get_inactive_hand() != src)
+				H << "<span class='danger'>Better luck next season, ball hog!</span>"
+				fail = TRUE
+				return
+			sleep(30)
```
Comment (Time-Green) [inline code/modules/holodeck/items.dm]: This sleep completely freezes the object, preventing any other procs from being called while it's active.
And the while checks for the slam being ready, wich is not changed in the loop, so I can only assume that happens in another proc. But other procs cannot be called for the object while in the loop...
How the fuck did this work on your test server?

Comment (1fbff5f83b23d39d38b1dfcb4cac8d9b) [reply code/modules/holodeck/items.dm]: slam_ready is changed to true by the spirit hoop when you click on it with the basketball in hand

Comment (Time-Green) [reply code/modules/holodeck/items.dm]: Oh yeah, that could work if it's ran by the hoop

**Comment by McDonald072** [comment]
Alright~~

So, to recap:
-To obtain a chaos ball you use a basketball on the exposed plutonium core.
-Activating gives you a popup asking if you are ready to jam.
-Once you begin jamming you will lose all items you are wearing, and gain athletic shorts and sweet jordans. You won't be able to equip anything but a radio to taunt others.
-A spirit hoop will spawn somewhere on the station. You and other people will start getting hints about where it is, similar to when clock cultists attempt to summon ratvar.
-You can smash walls, windows. grilles, windoors and tables apart, and stun people for a few seconds with the ball.
-Upon using the ball on the spirit hoop you will deliver the dunk of the millennium like it works now, but much stronger (25/60/150 explosion)

Any questions and or suggestions?

File: code/modules/holodeck/items.dm [hunk 3]
```diff
@@ -132,73 +168,100 @@
 	if(!instant_dunk)
 		if(alert("Are you ready to jam?",,"Let's slam!","Let's not!") != "Let's slam!")
 			return
+		if(blobstart.len > 0)
+			var/turf/targetturf = get_turf(pick(blobstart))
+			hoop = new /obj/structure/holohoop/spirit(targetturf)
+		else
+			H << "bit of a fuckup here"
```
Comment (ShadowDeath6) [inline code/modules/holodeck/items.dm]: ,':^)?

Comment (McDonald072) [reply code/modules/holodeck/items.dm]: that will never happen

**Comment by Kmc2000** [comment]
No credit?

Wow rude...

:^(

**Comment by McDonald072** [comment]
image

**Comment by LarissaMayrink** [comment]
Can't wait to be able to dunk

**Comment by Super3222** [comment]
I'll test this

**Comment by Super3222** [comment]
It works, and it's pretty good.



Here's the result of my testing. This was from a chaos dunk in the kitchen. The explosion is still expanding.

Actually, by the time I came back to submit this post it finally reached the vacant office.

**Comment by McDonald072** [comment]
Well, I think this is merge ready

**Comment by Super3222** [comment]
Make a changelog please

:cl: McDonald
rscadd: blah blah blah
/:cl:

**Comment by Cruix** [comment]
The spirit hoops do not disappear when you drop the ball while it is active, and you can still dunk it in them
You can also spam announcements by activating the ball over and over again

**Comment by 1fbff5f83b23d39d38b1dfcb4cac8d9b** [comment]
This message was created automatically by mail delivery software at Junk Email Filter. 

The message from "Cruix [Masked]" <FWD-737QTYSMENAIYCQANIQDAMKHLADIEAHYAZKGBJQA2ACQOIRIHRQAZYSAOYBCAOBCGCQE7GQEWU3KIAJYKPSS2JB4SJUZEICFEB7FGKRSPFSCQCGJFLLGAIRUWNAL2AGGBBMFKPC6CQECACVEIKQAOYAOQAEYAAYAAGAAWBLJSMAF2QCNIA3Q4LQ4LUKX6JPEAAQAAIA=@opayq.com> is being returned by server opayq-outbound.junkemailfilter.com on Fri, 09 Jun 2017 10:29:03 -0700

A message that you sent could not be delivered to all of its recipients.

NOTE: And this is important - Junk Email Filter is a front end spam filtering service. We accept email for our customers, filter it, and pass the good email on. Sometimes when we forward email as good the server we forward to rejects it. Generally if you are getting this message it is because the server we forwarded to  * NOT JUNK EMAIL FILTER * has rejected the message.

If you see below the phrase "error from remote mail server" then it is the server who we are forwarding to, NOT US who is rejecting the email. Read the reason carefully and see if you can correct the cause of the rejection.

The following address(es) failed:

  michal.sul@wp.pl
    host mx5.wp.pl [212.77.101.13]
    SMTP error from remote mail server after MAIL FROM:<FWD-737QTYSMENAIYCQANIQDAMKHLADIEAHYAZKGBJQA2ACQOIRIHRQAZYSAOYBCAOBCGCQE7GQEWU3KIAJYKPSS2JB4SJUZEICFEB7FGKRSPFSCQCGJFLLGAIRUWNAL2AGGBBMFKPC6CQECACVEIKQAOYAOQAEYAAYAAGAAWBLJSMAF2QCNIA3Q4LQ4LUKX6JPEAAQAAIA=@opayq.com>:
    502 unimplemented (#5.5.1)

File: code/modules/holodeck/items.dm [hunk 4]
```diff
@@ -92,6 +92,55 @@
...
+/obj/item/toy/beach_ball/holoball/chaos/New()
+	..()
+	notify_ghosts("Everybody get up, it's time to slam now!", source = src, action = NOTIFY_ORBIT)
+	poi_list |= src
+
+/obj/item/toy/beach_ball/holoball/chaos/Destroy()
+	poi_list -= src
+	..()
+
+/obj/item/toy/beach_ball/holoball/chaos/attack(atom/T, mob/user)
```
Comment (Iamgoofball) [inline code/modules/holodeck/items.dm]: no Chaos Dunk support?

get the fuck out :-1:

**Comment by 1fbff5f83b23d39d38b1dfcb4cac8d9b** [comment]
This message was created automatically by mail delivery software at Junk Email Filter. 

The message from "Iamgoofball [Masked]" <FWD-737QTYSMENAIYCQANIQDAMKHLADIEAHYAZIBBYDGNQCQABHABD4TMIBYE4YXABNFFYUARTRGXNICI3VJFJGV5KAVSQKQQFBCLCUQV4SWJFBLSAMSAISDCM2YGJYKMF36IBZQIPAHKEFIIHBIAAVEEYSAE4QC4MBJGARHAIJQFJYUGONAD3QBNYALVYG2OASAOI4ERIAAEA======@opayq.com> is being returned by server opayq-outbound.junkemailfilter.com on Fri, 09 Jun 2017 10:48:03 -0700

A message that you sent could not be delivered to all of its recipients.

NOTE: And this is important - Junk Email Filter is a front end spam filtering service. We accept email for our customers, filter it, and pass the good email on. Sometimes when we forward email as good the server we forward to rejects it. Generally if you are getting this message it is because the server we forwarded to  * NOT JUNK EMAIL FILTER * has rejected the message.

If you see below the phrase "error from remote mail server" then it is the server who we are forwarding to, NOT US who is rejecting the email. Read the reason carefully and see if you can correct the cause of the rejection.

The following address(es) failed:

  michal.sul@wp.pl
    host mx5.wp.pl [212.77.101.13]
    SMTP error from remote mail server after MAIL FROM:<FWD-737QTYSMENAIYCQANIQDAMKHLADIEAHYAZIBBYDGNQCQABHABD4TMIBYE4YXABNFFYUARTRGXNICI3VJFJGV5KAVSQKQQFBCLCUQV4SWJFBLSAMSAISDCM2YGJYKMF36IBZQIPAHKEFIIHBIAAVEEYSAE4QC4MBJGARHAIJQFJYUGONAD3QBNYALVYG2OASAOI4ERIAAEA======@opayq.com>:
    502 unimplemented (#5.5.1)

**Comment by Iamgoofball** [comment]
what the hell

**Comment by Cruix** [comment]
Someone in this chat is secretly a robot
It could be Iamgoofball
or Super3222
or even 1fbff5f83b23d39d38b1dfcb4cac8d9b
but I think it's Iamgoofball

**Comment by 1fbff5f83b23d39d38b1dfcb4cac8d9b** [comment]
I don't know what the fuck happened there

**Comment by Mrcreepssss** [comment]
hmmmm i think, there should be a way to escape the chaos dunk without stopping it, much ike with the SM cascade?

**Comment by fonedsky** [comment]
There is: you run as fast as possible away from the spirit hoop.

**Comment by Super3222** [comment]
As long as this is still admin-only I think it's good to go. I think there should also be a wizard event that can spawn down one of those chaos balls.

**Comment by 1fbff5f83b23d39d38b1dfcb4cac8d9b** [comment]
Understandable, have a nice day

**Pull request closed without merging**